gradle-suppressible-error-prone/src/main/java/com/palantir/gradle/suppressibleerrorprone/modes/Modes.java

```diff
@@ -0,0 +1,142 @@
+import javax.inject.Inject;
+import one.util.streamex.EntryStream;
+import one.util.streamex.StreamEx;
+import org.gradle.api.provider.Provider;
+import org.gradle.api.provider.ProviderFactory;
+import org.gradle.api.tasks.compile.JavaCompile;
+
+/**
+ * Use this class to determine the final combination of {@link Mode}s and {@link CommonOptions} for the various
+ * flags given to suppressible-error-prone.
+ */
+public abstract class Modes {
+    @Inject
+    protected abstract ProviderFactory getProviderFactory();
+
+    private final Map<ModeName, Mode> modes = Map.of(
+            ModeName.APPLY, new ApplyMode(),
+            ModeName.DISABLE, new DisableMode(),
+            ModeName.REMOVE_ROLLOUT, new RemoveRolloutMode(),
+            ModeName.SUPPRESS, new SuppressMode(),
+            ModeName.TIMINGS, new TimingsMode());
+
+    private final Set<ModeInterference> interferences = Set.of(
+            new DisableModeInterference(),
+            new RemovingAndSuppressingInterference(),
+            new SuppressingAndApplyingInterference());
+
+    public final CombinedValue modifyCheckApi() {
+        return ModifyCheckApiOption.combine(modesEnabledAndFlagValues().keySet().stream()
+                .map(modes::get)
+                .map(Mode::modifyCheckApi)
+                .collect(Collectors.toSet()));
+    }
+
+    public final CommonOptions commonOptionsFor(JavaCompile javaCompile) {
+        Map<ModeName, Optional<String>> modeNameToFlagValue = modesEnabledAndFlagValues();
+
+        Map<Interference, ModeInterference> interferingModes = StreamEx.of(interferences)
+                .mapToEntry(interference -> interference.interferesWith(modeNameToFlagValue.keySet()))
+                .filterValues(modeInterferenceResult -> modeInterferenceResult instanceof Interference)
+                .mapValues(modeInterferenceResult -> (Interference) modeInterferenceResult)
+                .invert()
+                .toMap();
+
+        Map<ModeName, CommonOptions> commonOptionsPerMode = EntryStream.of(modeNameToFlagValue)
+                .mapToValue((flagName, flagValue) -> {
+                    Mode mode = modes.get(flagName);
+                    return mode.configureAndReturnCommonOptions(new ModeOptionContext(flagValue, javaCompile));
+                })
+                .toMap();
+
+        Map<Interference, CommonOptions> commonOptionsAfterInterfering = EntryStream.of(interferingModes)
+                .mapToValue((interference, modeInterference) -> {
+                    return modeInterference.interfere(StreamEx.of(interference.interferingModes())
+                            .mapToEntry(commonOptionsPerMode::get)
+                            .toMap());
+                })
+                .toMap();
+
+        Set<ModeName> allInterferingModes = interferingModes.keySet().stream()
+                .flatMap(interference -> interference.interferingModes().stream())
+                .collect(Collectors.toSet());
```

Comment:
nit: can we just use `StreamEx` throughout?


```suggestion
        Set<ModeName> allInterferingModes = StreamEx.of(interferingModes.keySet())
                .flatMap(interference -> interference.interferingModes().stream())
                .toSet();
```

Reply:
We aren't using any streamex specific methods though? Is this just for better consistency? I kinda like avoiding using it unless there's a good reason to, streamex have had a number of abi breaks before and the more usages the more we risk hitting them.

Reply:
I have used streamex throughout.